src/commit.h

```diff
@@ -37,4 +37,11 @@ void git_commit__free(void *commit);
 int git_commit__parse(void *commit, git_odb_object *obj);
 int git_commit__parse_raw(void *commit, const char *data, size_t size);
 
+typedef enum {
+	GIT_COMMIT_PARSE_QUICK = (1 << 0), /**< Only parse parents and committer info */
+	GIT_COMMIT_PARSE_ALL   = (1 << 1), /**< Parse all commit data */
```

Comment:
So... do these two flags really make sense? After all, you're not using `PARSE_QUICK` anywhere in  `git_commit__parse_internal` and in fact `PARSE_QUICK` and `PARSE_ALL` do not make any sense together, right? I guess having only one of them would be more sensible, and I lean towards retaining `PARSE_QUICK`. In that case, default behaviour would be to parse everything and by passing the optional `PARSE_QUICK` flag we would skip certain information